src/bls/bls_aggregator.cpp

```diff
@@ -591,183 +663,262 @@ bls_rewards_response bls_aggregator::rewards_request(const address& addr, uint64
                 service_node_list, core.l2_tracker(), result.signers_bls_pubkeys);
     }
 
-    // NOTE: Store the response in to the cache if the number of non-signers was less than 1/3rd.
+    // NOTE: Store the response in to the cache if the number of non-signers is small enough to
+    // constitute a valid signature.
     uint64_t non_signers_count = total_requests - result.signers_bls_pubkeys.size();
-    if (non_signers_count < (total_requests / 3)) {
-        std::lock_guard lock{mutex};
+    if (non_signers_count <= contract::rewards_bls_non_signer_threshold(total_requests)) {
+        std::lock_guard lock{rewards_response_cache_mutex};
         rewards_response_cache[addr] = result;
     }
 
     return result;
 }
 
-void bls_aggregator::get_removal(oxenmq::Message& m) {
-    oxen::log::trace(logcat, "Received omq removal signature request");
-    bls_removal_request removal_request = extract_removal_request(m);
-    if (!removal_request.good)
+void bls_aggregator::get_exit_liquidation(oxenmq::Message& m, bls_exit_type type) const {
+    oxen::log::trace(logcat, "Received omq {} signature request", bls_exit_type_label(type));
+    bls_exit_request request = extract_exit_request(m);
+    if (!request.good)
         return;
 
-    // right not its approving everything
-    if (!core.is_node_removable(removal_request.remove_pk)) {
+    bool removable = true;
```

Comment:
Very minor:
```suggestion
    bool removable;
```
because it should *always* get set in the following switch; if it isn't defaulted and somehow doesn't get set in the switch because of some future change, there would be useful compiler warnings about using an uninitialized variable that won't show up when it has a default.

Reply:
Turns out this suggestion was wrong: it seems an exhaustive list of all possible enum values in a switch isn't enough to convince gcc that the value is initialized and it produced uninitialized warnings.  I've pushed a fix to revert this.